### ISSUE TITLE
Porngo(YesPornPlease) quality issue.

### ISSUE THREAD

**Issue by imurreflection** [dobbelina/repository.dobbelina#337]
It appears that even if the video is marked as 720p, it still plays at 544p resolution which unfortunately is quite low resolution. Can we please fix this

**Comment by jdoedev123** [comment]
I am always getting 1280x720 when 720p selected.
Is their a particular Video that was 544p? Kodi sometimes marks as 720p some resolutions between 480p and 720p.

**Comment by imurreflection** [comment]
All the videos are 544p even if they are marked as 720p. I have checked the same on website but there is no quality selection available so cannot confirm.  I am using KM9 Pro running Android 10.

**Comment by jdoedev123** [comment]
Running Kodi 18.6 with Confluence interface. For example "POV Casting 720p 1:17:09" shows as 1280x720.
Details from log:  

https://www.porngo.com/get_file/1014/49372fa1a1a8b8d03d3095e019235db11c33c80e78/658000/658979/658979_720m.mp4/|verifypeer=false

Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 1848 kb/s, 25 fps, 25 tbr, 12800 tbn, 50 tbc (default)

**Comment by imurreflection** [comment]
Running Kodi 18.9 Leia with Aeon Nox Silvo on Android TV Box KM9 Pro with 4 GB Ram. Getting every video resolution as 480P (Just observed). Checked the same video and found it displaying quality resolution as 480p even when it it marked as 720P. Please see the screenshots.
![IMG_20201122_171844](https://user-images.githubusercontent.com/30263396/99903114-53821180-2ce8-11eb-8231-8ff342435890.jpg)
![IMG_20201122_171908](https://user-images.githubusercontent.com/30263396/99903118-554bd500-2ce8-11eb-8302-337679871778.jpg)
![IMG_20201122_172006](https://user-images.githubusercontent.com/30263396/99903119-567d0200-2ce8-11eb-9a08-3c2b5b4d68f2.jpg)
![IMG_20201122_172107](https://user-images.githubusercontent.com/30263396/99903120-57159880-2ce8-11eb-9fe0-3352c0602482.jpg)

**Comment by jdoedev123** [comment]
Hmm on my Tablet running Android 7.0 with Kodi 18.9 and UWC 1.2.33 the same video runs as 1280x720. My CPU/GPU is EXYNOS 5433 which has a Mali-T720 GPU. System Memory usage is about 89& of about 3 GB.
I am unsure where your device is picking up the lower resolution stream. On my Desktop Browser only a link to the 720p link is in the video page scraped for a playable stream.  I thought it possible Android was receiving a different video page but my Tablet apparently does not so I can't test easily.

**Comment by imurreflection** [comment]
Even I am not sure about this. This is actually happening since the day the site was added. Let's see if someone else can look into this. This is actually a very good site and I just don't want to watch it in lower resolution. Perhaps @camilt has required hardware to check this anamoly.

**Comment by imurreflection** [comment]
@camilt. May I request you to kindly look at this issue please?

**Comment by frankiek3** [comment]
The issue is with the site not the add-on. The porngo link will resolve to a content delivery network (CDN) link at the site's discretion. No guaranteed resolution.

**Comment by imurreflection** [comment]
I agree @frankiek3 but is this limitation applicable only on Android TV boxes?  Is there any manual changes we can perform in current url resolver. Kindly reply

**Comment by jdoedev123** [comment]
@imurreflection  I am guessing we could try to fool the system by spoofing the header to impersonate another type of Desktop system requesting content. This is actually a common technique in UWC. However, I don't have a setup to properly test. My Nvidia Shield TV and Fire TV died (along with a TV) due to a bad HDMI cable short. I just bought some Fire TV Sticks but have not installed Kodi on them and I am busy with converting UWC to Kodi 19.

**Comment by imurreflection** [comment]
Can you please advise how to fool this system. I am experiencing this issue with some more sites. May be, this could resolve the issue for all. Thanks mate @jdoedev

**Comment by jdoedev123** [comment]
@imurreflection  I can not guarantee anything without a Android TV test setup, but try:
porngo.py about line 99:      
OLD videopage = utils.getHtml(url, '')    
NEW videopage = utils.getHtml3(url)

All I can tell is it does not break the Desktop version, but I can not confirm it will fix your problem.

**Comment by imurreflection** [comment]
Hi mate @jdoedev123.  I tried your method and modified py file but unfortunately it has no effect and all the stream still plays at 480p. It's such a pity thing for this amazing site. I requested @camilt to look into this twice as you don't have the required setup but perhaps he is busy. If he or anyone else can provide some input that would b great. I really appreciate your sincere efforts here. Thanks again
